### PR TITLE
626 upgrade to python 11

### LINKED ISSUE #626: Upgrade to Python 11
### Motivation / Current Behaviour

Python says version 11 is 10-60% faster, we should take that

### Enhancement description

- [x] Test if python 11 runs our programm
- [x] quick test if 10-60% performance increase is really achieve
- [ ] fix conflicts

### Additional context

_No response_

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project

### PR DESCRIPTION
**Pull request by MaxBetzDLR**
## Changes

Please briefly list the changes made:

[Describe here in 2-3 lines. For details, reference to issues to avoid redundant information.]

## Merge Request - GuideLine Checklist

- *Check our [git workflow](https://github.com/DLR-SC/memilio/wiki/git-workflow) before opening a Pull request/Merge request.*
- *Request a reviewer when your work is ready to review, before this please use the draft feature.*

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/DLR-SC/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added

### Checks by code reviewer(s)

- [x] Corresponding issue(s) is/are linked and addressed
- [x] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [x] Appropriate **unit tests** have been added, CI passes and code coverage is acceptable (did not decrease)
- [x] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)

### Additional context

Please list additional information or things a reviewer should look out for.

Closes #626

### CODE REVIEW

File: pycode/memilio-epidata/requirements-dev.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # dev dependencies, pyfakefs 4.2.0 is broken!
-pyfakefs==4.1.0
+pyfakefs>=4.2.1
 coverage>=7.0.1
-pylint<=2.11.1
-pylint_json2html<=0.3.0
+pylint>=2.13.0,<2.16
+pylint_json2html==0.4.0
```
Comment (patricklnz) [inline pycode/memilio-epidata/requirements-dev.txt:5]: Update comments why we use this versions

File: pycode/memilio-epidata/setup.py [hunk 2]
```diff
@@ -90,8 +90,8 @@ def run(self):
             'pyfakefs>=4.2.1',
             # coverage 7.0.0 can't find .whl files and breaks CI
             'coverage>=7.0.1',
-            'pylint<=2.11.1',
-            'pylint_json2html<=0.3.0',
+            'pylint>=2.13.0,<2.16',
+            'pylint_json2html==0.4.0',
```
Comment (patricklnz) [inline pycode/memilio-epidata/setup.py:94]: Comments should be the same as in requirements-dev.txt

File: pycode/memilio-plot/requirements-dev.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # dev dependencies, pyfakefs 4.2.0 is broken!
-pyfakefs==4.1.0
-coverage
-pylint<=2.11.1
-pylint_json2html<=0.3.0
+pyfakefs>=4.2.1
+coverage>=7.0.1
+pylint>=2.13.0,<2.16
+pylint_json2html==0.4.0
```
Comment (patricklnz) [inline pycode/memilio-plot/requirements-dev.txt:5]: see above

File: pycode/memilio-plot/setup.py [hunk 4]
```diff
@@ -81,9 +81,9 @@ def run(self):
         'dev': [
             # smaller pyfakefs versions use deprecated functions for matplotlib versions >=3.4
             'pyfakefs>=4.2.1',
-            'coverage',
-            'pylint<=2.11.1',
-            'pylint_json2html<=0.3.0',
+            'coverage>=7.0.1',
+            'pylint>=2.13.0,<2.16',
+            'pylint_json2html==0.4.0',
```
Comment (patricklnz) [inline pycode/memilio-plot/setup.py:86]: see above

**Review by patricklnz — CHANGES_REQUESTED** [review]
The epidata test for hospilization data fails with a RecursionError, this has to be fixed. Also some comments on why we use specific versions. Else looks good to me.

**Comment by dabele** [comment]
pybind was upgraded here: #705

**Review by patricklnz — APPROVED** [review]
Looks good to me.

File: pycode/memilio-generation/memilio/generation/utility.py [hunk 5]
```diff
@@ -25,7 +25,15 @@
 import subprocess
 from typing import Any, List, TextIO
 
-import importlib.resources
+
+import sys
+
+if sys.version_info >= (3, 9):
+    # For python 3.8 and newer
```
Comment (mknaranja) [inline pycode/memilio-generation/memilio/generation/utility.py:32]:
```suggestion
    # For python 3.9 and newer
```
?

**Review by mknaranja — APPROVED** [review]

**Comment by mknaranja** [comment]
Thank you @MaxBetzDLR and @patricklnz. Good work!

**Pull request merged**